inc/thorvg.h

```diff
@@ -232,10 +232,20 @@ class TVG_EXPORT Paint
      * @param[in] m The 3x3 augmented matrix.
      *
      * @return Result::Success when succeed, Result::FailedAllocation otherwise.
-     *
      */
     Result transform(const Matrix& m) noexcept;
 
+    /**
+     * @brief Gets the matrix of the affine transformation of the object.
+     *
+     * @param[out] m The 3x3 augmented matrix.
+     *
+     * @retval Result::Success when succeed.
+     * @retval Result::InvalidArguments If a @c nullptr is passed as the argument.
+     * @retval Result::InsufficientCondition In case no transformation was set.
+     */
+    Result transform(const Matrix** m) const noexcept;
```

Comment:
the internal matrix data is very fragle.. (it can be freed or values are changed anytime) 
Thus I suggest either way:
1. Matrix transform();   
2. Result transform(Matrix& m);

The first one is the best as an api  for me but I'm not sure the exact situation of your scenario... Such If you're trying to avoid data copy for optimization point.

Reply:
Also, you should not add const here because that paint data can be updated.

Reply:
@hermet please check now, I returned the matrix by val.
yes, you're right, I didn't want to copy the data, but I think in this case it's ok

I removed const. btw - is the composite API in thorvg.h - is there `const` ok?

Reply:
@mgrudzinska no, it's wrong :-@